### PR TITLE
Docs: Update release process

### PR DESCRIPTION
**Pull request by nzakas**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix ([template](https://github.com/eslint/eslint/blob/master/templates/bug-report.md))
[ ] New rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

*If the item you've check above has a template, please paste the template questions here and answer them.*

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [ ] I've included tests for my change
- [x] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**

Updated the maintainer guide release docs to better spell out the release process.

**Is there anything you'd like reviewers to focus on?**

Is the information clear?

### CODE REVIEW

File: docs/maintainer-guide/releases.md [hunk 1]
```diff
@@ -5,41 +5,44 @@ Releases are when a project formally publishes a new version so the community ca
...
-You can run a regular release with the following command:
+1. Review open pull requests to see if any should be merged. In general, you can merge pull requests that:
+  * Have been open at least two days and have been reviewed (these are just waiting for merge).
+  * Important pull requests (as determined by the team). You should stop and have people review before merging if they haven't been already.
+  * Documentation changes.
+  * Small bug fixes written by a team member.
+2. Log into Jenkins and schedule a build for the "ESLint Release" job.
+3. Wait for the "ESLint Release" job to complete.
+4. Update the release blog post with a "Highlights", including new rules and anything else that's important.
+4. Make release announcement in the chatroom.
```
Comment (platinumazure) [inline docs/maintainer-guide/releases.md]: Two \#4 items.

**Comment by platinumazure** [comment]
I think you'll have markdownlint problems because some sub-bullets are only indented 2 spaces (we have it configured for 4-space sub-list indents).

**Comment by nzakas** [comment]
Ah crap, right. I'll fix that.

**Comment by platinumazure** [comment]
LGTM, thanks for updating this!

**Comment by btmills** [comment]
LGTM

**Comment by kaicataldo** [comment]
LGTM

**Comment by gyandeeps** [comment]
LGTM :+1:

**Pull request merged**